model-engine/model_engine_server/inference/domain/gateways/streaming_storage_gateway.py

```diff
@@ -8,7 +8,7 @@ class StreamingStorageGateway(ABC):
     """
 
     @abstractmethod
-    def put_record(self, stream_name: str, record: Dict[str, Any]) -> None:
+    def put_record(self, stream_name: str, record: Dict[str, Any]) -> Any:
```

Comment:
nit: it looks like we're implicitly assuming this returns a Dict with a key `ResponseMetadata` where we use the return value, looks like `TypedDict` doesn't quite work since there might be extra values returned, could we have this return a Dict type (or otherwise constrain the return type?)